sherpa/astro/data.py

```diff
@@ -1590,12 +1612,18 @@ def __init__(self, name, channel, counts, staterror=None, syserror=None,
                  bin_lo=None, bin_hi=None, grouping=None, quality=None,
                  exposure=None, backscal=None, areascal=None, header=None):
 
+        # Set the size of the object as soon as we know (it makes it
+        # easier to get usable error messages when checking the
+        # related fields). This is only done for the channel case to
+        # allow the counts field to be set but the channel field
+        # unset.
+        #
         channel = _check(channel)
+        if channel is not None:
+            self._size = len(channel)
```

Comment:
For all-but-PHA we have the number of elements as the size. For PHA data, do we want the size to be the number of channels or the number of grouped bins?

Reply:
I suggest to use the number of channels as size for consistency.

Reply:
This has now been added to the `Data` class: it uses the size attribute (if set, otherwise returns 0), so ifor PHA data it does use the number of channels (not groups).